lib/gui/pages/main/controllers/flash.js

```diff
@@ -100,8 +100,8 @@ module.exports = function (
       } else if (error.code === 'ENOSPC') {
         FlashErrorModalService.show(messages.error.notEnoughSpaceInDrive())
       } else {
-        FlashErrorModalService.show(messages.error.genericFlashError())
-        exceptionReporter.report(error)
+        FlashErrorModalService.show(`${messages.error.genericFlashError()}\n\n${error}`)
+        analytics.logException(error)
```

Comment:
I don't fully understand this code, or why this change is necessary, but given that `exceptionReporter` is also used in multiple other places, would a more-comprehensive solution to be add some kind of 'wrapper' around the `osDialog.showError` call (if that's indeed where this is falling over), so that you can detect if a native dialog is being displayed, and not trigger the timeout-error if so?
(And again, apologies if I'm asking stupid questions)

Reply:
No worries man, they're not stupid questions at all. I feel like since Windows is the only OS with a dialog that has this effect we'd be maintaining a non-trivial workaround just for Windows, when we instead have dialogs that don't block.

That is indeed how it used to work before

Reply:
So are we sure that none of the other uses of [exceptionReporter](https://github.com/resin-io/etcher/search?q=exceptionReporter&type=Code&utf8=%E2%9C%93) need workarounds similar to what you're doing here?

Is this potentially a bug in Electron that the native-dialog-code behaves differently on Windows than it does on OSX or Linux?
I can't remember the details or the reason why, but IIRC being able to use native dialogs was a "big thing" when @jviotti was first adding it? (Or possibly I'm getting confused with some other part)... Ah, looking at [CHANGELOG.md](https://github.com/resin-io/etcher/blob/master/CHANGELOG.md) it looks like I was thinking of the Update dialog rather than the Error dialog ;-)

> > the user would have seen two dialogs?

> That is indeed how it used to work before

Weird, any idea why it was done that way?